src/index.js

```diff
@@ -8,3 +10,32 @@ export {
 
 export * from './service';
 export * from './message';
+
+export default (config) => {
+  const internalLogger = new Logger(config);
+
+  return (label) => {
+    const labelledLogger = internalLogger.Label(label);
+
+    const logger = level => (...args) => {
+      let msg;
+
+      if (args.length === 2) {
```

Comment:
Yes it does, but with the single-method API we're sort of guessing what the user actually needs, and since we can have an implicit normal log with `message, { additionalInfo }` and an implicit error log with `message, Error`, we could only guess but the argument type or the log level (e.g. `.error()` means error message)

I would favor the type detection approach